apis/keda/v1alpha1/scaledjob_types_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+)
+
+func TestScaledJob(t *testing.T) {
+	tests := []struct {
+		name        string
+		expectedMax int64
+		expectedMin int64
+		maxReplicas *int32
+		minReplicas *int32
+	}{
+		{
+			name:        "MaxReplicaCount is set to 10 and MinReplicaCount is set to 1",
+			expectedMax: 9,
+			expectedMin: 1,
+			maxReplicas: int32Ptr(10),
+			minReplicas: int32Ptr(1),
+		},
+		{
+			name:        "MaxReplicaCount is set to 0 and MinReplicaCount is set to 1",
+			expectedMax: 100,
+			expectedMin: 0,
+			maxReplicas: int32Ptr(0),
+			minReplicas: int32Ptr(1),
+		},
+		{
+			name:        "MaxReplicaCount is nil and MinReplicaCount is set to 1",
+			expectedMax: 100,
+			expectedMin: 0,
+			maxReplicas: nil,
+			minReplicas: int32Ptr(1),
+		},
```

Comment:
Could you add another test case like this but setting minReplicas to `nil`?

Reply:
yeah, refactored the test a little